consent_manager/api.rst

```diff
@@ -0,0 +1,250 @@
+
+Get consent types
+`````````````````````````````````````
+Fetches a list of consent types.
+
+Code::
+
+    ppms.cm.api('getComplianceTypes', onFulfilled, onRejected);
+    dataLayer.push({'event': 'ppms.cm:getComplianceTypes', parameters: [onFulfilled, onRejected]});
+
+.. function:: onFulfilled(types)
+
+    The fulfilment handler callback (called with result). This function is **required**.
+
+    :param Array<string> types: **Required** Array of consent types
+
+        Example::
+
+            ["remarketing", "analytics"]
+
+.. function:: onRejected(error)
+
+    The rejection handler callback (called with error code). If not specified, exception will be thrown in main stacktrace.
+
+    :param string|object error: **Required** Error code or exception.
+
+Get new consent types
+`````````````````````````````````````
+Fetches a list of new consent types which were appearing after given consents.
+
+Code::
+
+    ppms.cm.api('getNewComplianceTypes', onFulfilled, onRejected);
+    dataLayer.push({'event': 'ppms.cm:getNewComplianceTypes', parameters: [onFulfilled, onRejected]});
+
+.. function:: onFulfilled(types)
+
+    The fulfilment handler callback (called with result).
+
+    :param Array<string> types: **Required** Array of consent types
+
+        Example::
+
+            ["remarketing", "analytics"]
+
+.. function:: onRejected(error)
+
+    The rejection handler callback (called with error code).
+
+    :param string|object error: **Required** Error code or exception.
+
+
+Set compliance settings
+`````````````````````````````````````
+Set compliance settings base on user decision.
+This API command might be useful when user interact with custom, extended UI that react on user approve/reject action.
+After successful, Consent Manager internally send consent settings to tracking server and force page view on tags.
+
+Code::
+
+    ppms.cm.api('setComplianceSettings', settings, onFulfilled, onRejected);
+    dataLayer.push({'event': 'ppms.cm:setComplianceSettings', parameters: [settings, onFulfilled, onRejected]});
+
+.. object:: settings
+
+    The consent settings object.
+
+        Example::
+
+            {consents: {analytics: {status: -1}}}
+
+    Where `consent.analytics` is consent type and status indicate:
+
+    * `-1` - user does not interact, e.q. close consent popup without any decision
+    * `0` - user has rejected the consent
+    * `1` - user has approved the consent
+
+.. function:: onFulfilled()
+
+    The fulfilment handler callback. This function is **required**.
```

Comment:
I think that in other files we use standard where we declare if param is required or optional at the start of description. It may be better to stick to that.

`**Required** The fulfilment handler callback.`

Reply:
https://raw.githubusercontent.com/PiwikPRO/PPMS-PublicDocs/master/audience_manager/form-tracker.rst - more detailed example here

Reply:
Done.